src/ocean_emulators/config.py

```
@@ -287,31 +295,89 @@ def build(
         )
 
 
-class EncoderConfig(BaseConfig):
-    patch_size: int | list[int] = Field(
-        default=4,
-        description="Either a square patch (int) or a rectangular patch of [height: int, width: int]. It must evenly divide the grid size.",
-    )
-    perceiver_depth: int = 6
-    perceiver_latent_dim: int = Field(
+PerceiverImpl = Literal["auto", "naive", "flash"]
+
+
+class PerceiverConfig(BaseConfig):
+    """A standard config interface to various perceiver implementations.
+
+    The `implementation="auto"` option will guess what is the best implementation given the runtime environment.
+    """
+
+    implementation: PerceiverImpl = "auto"
+    depth: int = 6
+    latent_dim: int = Field(
         default=128,
         description="The small, latent dimension of the Perceiver. This is the `N` dimension for the Perceiver's `O(M*N)` complexity",
     )
-    perceiver_num_latents: int = Field(
+    num_latents: int = Field(
         default=512,
         description="The number of latent vectors in the Perceiver. This is the `M` dimension for the Perceiver's `O(M*N)` complexity",
     )
 
+    def build(
+        self, in_channels: int, out_channels: int, patch_size: tuple[int, int]
+    ) -> nn.Module:
+        # This is not really a "frequency" but a maximum of the width appears to be reasonable from looking at the code.
+        max_freq = max(*patch_size)
+
+        # TODO(alxmrs,jder): Each implementation takes the mean of the num_latents dim to produce the final output_dim.
+        #  Why compute the mean? Is it better to directly project from the num_latents x latent_dim?
+        if (
+            self.implementation == "auto" and FLASH_ENABLED
+        ) or self.implementation == "flash":
```

@alxmrs I think the behavior here is that if you forget to include `--extras cuda` it will silently fall back to the naive implementation. Can you please change this so that "auto" always means "flash" if we're using a torch cuda backend (ie so you get an error if you try to use it on GPU without installing the cuda extras)